src/spikeinterface/sorters/basesorter.py

```diff
@@ -145,9 +145,10 @@ def initialize_folder(cls, recording, output_folder, verbose, remove_existing_fo
         elif recording.check_serializability("pickle"):
             recording.dump(output_folder / "spikeinterface_recording.pickle", relative_to=output_folder)
         else:
-            # TODO: deprecate and finally remove this after 0.100
-            d = {"warning": "The recording is not serializable to json"}
-            rec_file.write_text(json.dumps(d, indent=4), encoding="utf8")
+            raise RuntimeError(
+                "This recording is not serializable and so can not be sorted. Consider `recording.save()` to save a "
+                "compatible binary file."
+            )
```

Comment:
I think we should keep the possibility of running a sorting on a recording living only in memory.
And so no dump into a file.
This could be usefull for benchmark of simulated in memeory.
So I would `pass` silently no ?

Reply:
This was inspired by the issue I linked where we asked you to comment. fine to let it pass if you're okay with it, but do you want to read that issue and comment there or here on your opinions about it.

Reply:
Based on your comment in the issue what I would argue is we add this runtime error but with a message like:

```
This recording is not serializable and so can not be sorted. Consider recording.save() to generate a compatible binary file. We plan to support in memory sorting for some sorters in the future
```

Reply:
Sam and I discussed this and he is okay with the error for now. Any opinions on adding either 

1) a comment saying we should support in-memory in the future
2) saying in the error message that we plan to support in the future

@alejoe91

Reply:
No I think it's fine to as is :)